cargo-public-api/src/error.rs

```diff
@@ -1,5 +1,63 @@
+use public_api::{diff::ChangedPublicItem, PublicItem};
+
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
-    #[error("The API diff is not allowed as per --deny")]
-    DiffDenied,
+    #[error("The API diff is not allowed as per --deny: {0}")]
+    DiffDenied(Violations),
+}
+
+#[derive(Debug)]
+pub struct Violations {
+    /// These items were added to the API, but no items may be added to the API
+    added: Vec<PublicItem>,
+
+    /// These items were changed in the API, but no items may be changed in the API
+    changed: Vec<ChangedPublicItem>,
+
+    /// These items were removed from the API, but no items may be removed from the API
+    removed: Vec<PublicItem>,
+}
+
+impl Violations {
+    pub fn new() -> Violations {
+        Violations {
+            added: Vec::new(),
+            changed: Vec::new(),
+            removed: Vec::new(),
+        }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.added.is_empty() || self.changed.is_empty() || self.removed.is_empty()
```

Comment:
I tried this:

```bash
./scripts/create-test-git-repo.sh /tmp/foo
cargo run -- --manifest-path /tmp/foo/Cargo.toml --diff-git-checkouts v0.1.0 v0.2.0 --deny=added
```

and noticed it didn't fail. Debugged it a bit and noticed this is wrong, it should be like this:

```suggestion
        self.added.is_empty() && self.changed.is_empty() && self.removed.is_empty()
```

:)

Reply:
Meh, thanks for catching that!